packages/flutter/lib/src/services/text_input.dart

```diff
@@ -678,14 +681,18 @@ class TextInputConnection {
   /// 2. [transform]: a matrix that maps the local paint coordinate system
   ///                 to the [PipelineOwner.rootNode].
   void setEditableSizeAndTransform(Size editableBoxSize, Matrix4 transform) {
-    SystemChannels.textInput.invokeMethod<void>(
-      'TextInput.setEditableSizeAndTransform',
-      <String, dynamic>{
-        'width': editableBoxSize.width,
-        'height': editableBoxSize.height,
-        'transform': transform.storage,
-      },
-    );
+    if (editableBoxSize != _cachedSize || transform != _cachedTransform) {
+      _cachedSize = editableBoxSize;
+      _cachedTransform = transform;
```

Comment:
Wondering if we should store a copy of `transform`. Unfortunately, `Matrix4` is mutable.

Reply:
done.  thanks for the suggestion